gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStubTest.java

```diff
@@ -0,0 +1,153 @@
+    HttpJsonOperationsStub.create(
+        ClientContext.newBuilder()
+            .setCredentials(NoCredentialsProvider.create().getCredentials())
+            .setDefaultCallContext(FakeCallContext.createDefault())
+            .build(),
+        new HttpJsonOperationsCallableFactory(),
+        TypeRegistry.newBuilder().build(),
+        ImmutableMap.of(
+            "google.longrunning.Operations.ListOperations",
+            HttpRule.newBuilder()
+                .setGet("testList")
+                .addAdditionalBindings(HttpRule.newBuilder().setGet("testList2"))
+                .build(),
+            "google.longrunning.Operations.GetOperation",
+            HttpRule.newBuilder()
+                .setGet("testGet")
+                .addAdditionalBindings(HttpRule.newBuilder().setGet("testGet2"))
+                .build(),
+            "google.longrunning.Operations.DeleteOperation",
+            HttpRule.newBuilder()
+                .setDelete("testDelete")
+                .addAdditionalBindings(HttpRule.newBuilder().setDelete("testDelete2"))
+                .build(),
+            "google.longrunning.Operations.CancelOperation",
+            HttpRule.newBuilder()
+                .setPost("testCancel")
+                .addAdditionalBindings(HttpRule.newBuilder().setPost("testCancel2"))
+                .build()));
+  }
+
+  @Test
+  public void testMethodDescriptorsURI() {
+    List<ApiMethodDescriptor> apiMethodDescriptorList =
+        HttpJsonOperationsStub.getMethodDescriptors();
+    assertThat(apiMethodDescriptorList.get(0).getRequestFormatter().getPathTemplate().toRawString())
+        .isEqualTo("testList");
+    assertThat(apiMethodDescriptorList.get(1).getRequestFormatter().getPathTemplate().toRawString())
+        .isEqualTo("testGet");
+    assertThat(apiMethodDescriptorList.get(2).getRequestFormatter().getPathTemplate().toRawString())
+        .isEqualTo("testDelete");
+    assertThat(apiMethodDescriptorList.get(3).getRequestFormatter().getPathTemplate().toRawString())
+        .isEqualTo("testCancel");
+  }
+
+  @Test
+  public void testMethodDescriptorsAdditionalBindings() {
+    // getAllMethodDescriptors() returns the MethodDescriptors in specific order
+    // The order is: List, Get, Delete, Cancel
+    List<ApiMethodDescriptor> apiMethodDescriptorList =
+        HttpJsonOperationsStub.getMethodDescriptors();
+    ProtoMessageRequestFormatter<ListOperationsRequest>
+        listOperationsRequestProtoMessageRequestFormatter =
+            (ProtoMessageRequestFormatter<ListOperationsRequest>)
+                apiMethodDescriptorList.get(0).getRequestFormatter();
+    assertThat(listOperationsRequestProtoMessageRequestFormatter.getAdditionalPathTemplates())
+        .hasSize(1);
+    assertThat(
+            listOperationsRequestProtoMessageRequestFormatter
+                .getAdditionalPathTemplates()
+                .get(0)
+                .toRawString())
+        .isEqualTo("testList2");
```

Comment:
Verifying a value that was set up in `setUp` method is also not a good practice, see [here](go/unit-testing-practices#setUp)

Reply:
This test tries to cover the `updateDefaultApiMethodDescriptors` function. Perhaps the easier/ cleaner way for this is to just open method as default scope and test it that way?